### ISSUE TITLE
TypeScript

### ISSUE THREAD

**Issue by TheKnarf** [digitalie/one-loader#45]
How to I set this up to work with TypeScript?

**Comment by MunGell** [comment]
Hi @TheKnarf 

You can set it up to work with Typescript like this:

In your component file:

```html
<script type="typescript">
...
</script>
```

In your `webpack.conf.js` file:

```
...
{
    test: /\.one$/,
    loader: 'one-loader',
    options: {
        map: {
            'text/scss': [MiniCssExtractPlugin.loader, 'css-loader?modules=true', 'sass-loader'],
            'typescript': 'ts-loader'
        }
    }
},
...
```

Make sure all required modules are installed as well.

If you want to compile Typescript with babel, change `.babelrc` file, like it is described here: https://medium.com/@francesco.agnoletto/how-to-set-up-typescript-with-babel-and-webpack-6fba1b6e72d5

Have a look at Example 02 to see how it works with SCSS styles (similar concept applies there): https://github.com/digitalie/one-loader/tree/master/examples/02_redux-todos

**Comment by TheKnarf** [comment]
Maybe we could add a TypeScript example?

**Comment by MunGell** [comment]
Absolutely, feel free to contribute an example with it - will be much appreciated!
